tests/support/types.ts

```diff
@@ -0,0 +1,112 @@
+import { Page } from '@playwright/test';
+
+export interface CheckBalanceUpdateType {
+  page: Page;
+  initialBalance: number;
+  expectedChange: number;
+}
+
+export interface VerifyBalanceChangeType {
+  page: Page;
+  initialBalance: number;
+  clickedButton: 'ping' | 'pong';
+}
+
+export interface CheckButtonStatusType {
+  page: Page;
+  type: import('./testdata').PingPongEnum;
+  lastClickedButton: 'ping' | 'pong' | null;
+}
+
+export interface HandlePingPongType {
+  page: Page;
+  type: import('./testdata').PingPongEnum;
+}
+
+export interface WaitForPageByUrlSubstringType {
+  page: Page;
+  urlSubstring: string;
+  timeout?: number;
+}
+
+export interface WaitForNewPageType {
+  page: Page;
+  urlSubstring: string;
+  timeout: number;
+}
+
+export interface CreateNotFoundErrorType {
+  urlSubstring: string;
+  timeout: number;
+  currentPageUrl: string;
+  availablePagesUrls: string;
+}
+
+export interface ConnectWebWalletType {
+  page: Page;
+  loginMethod: {
+    keystore?: string;
+    pem?: string;
+    password?: string;
+  };
+}
+
+export interface AuthenticateWithKeystoreType {
+  walletPage: Page;
+  keystorePath: string;
+  keystorePassword: string;
+}
+
+export interface AuthenticateWithPemType {
+  walletPage: Page;
+  pemPath: string;
+}
+
+export interface ConfirmWalletTransactionType {
+  page: Page;
+  loginMethod: {
```

Comment:
This seems to be repeated above, in the `ConnectWebWalletType` interface, any way we can extract this in a `LoginMethodType` interface and reference that in both cases?

Reply:
Awesome. 👍